subcommand/server-acl-init/policies.go

```diff
@@ -0,0 +1,158 @@
+package serveraclinit
+
+import (
+	"bytes"
+	"strings"
+	"text/template"
+)
+
+type rulesData struct {
+	EnableNamespaces               bool
+	ConsulSyncDestinationNamespace string
+	EnableSyncK8SNSMirroring       bool
+	SyncK8SNSMirroringPrefix       string
+}
+
+const snapshotAgentRules = `acl = "write"
+key "consul-snapshot/lock" {
+   policy = "write"
+}
+session_prefix "" {
+   policy = "write"
+}
+service "consul-snapshot" {
+   policy = "write"
+}`
+
+const entLicenseRules = `operator = "write"`
+
+func (c *Command) agentRules() (string, error) {
+	agentRulesTpl := `
+  node_prefix "" {
+    policy = "write"
+  }
+{{- if .EnableNamespaces }}
+namespace_prefix "" {
+{{- end }}
+  service_prefix "" {
+    policy = "read"
+  }
+{{- if .EnableNamespaces }}
+}
+{{- end }}
+`
+
+	return c.renderRules(agentRulesTpl)
+}
+
+func (c *Command) dnsRules() (string, error) {
+	// DNS rules need to have access to all namespaces
+	// to be able to resolve services in any namespace.
+	dnsRulesTpl := `
+{{- if .EnableNamespaces }}
+namespace_prefix "" {
+{{- end }}
+  node_prefix "" {
+     policy = "read"
+  }
+  service_prefix "" {
+     policy = "read"
+  }
+{{- if .EnableNamespaces }}
+}
+{{- end }}
+`
+
+	return c.renderRules(dnsRulesTpl)
+}
+
+// This assumes users are using the default name for the service, i.e.
+// "mesh-gateway".
+func (c *Command) meshGatewayRules() (string, error) {
+	// Mesh gateways can only act as a proxy for services
+	// that its ACL token has access to. So, in the case of
+	// Consul namespaces, it needs access to all namespaces.
+	meshGatewayRulesTpl := `
```

Comment:
I was testing something else and realized these acls are wrong. We need:
```
service "mesh-gateway" {
   policy = "write"
}
service_prefix "" {
   policy = "read"
}
node_prefix "" {
   policy = "read"
}
agent_prefix "" {
   policy = "read"
}
```

The mesh gateway is always registered only in the default namespace, so if using namespaces we need:
```
service_prefix "gateway" {
   policy = "write"
}
namespace_prefix "" {
  service_prefix "" {
     policy = "read"
  }
  node_prefix "" {
     policy = "read"
  }
  agent_prefix "" {
     policy = "read"
  }
}
```

Reply:
NOTE: I'm not sure if agent_prefix needs to be namespaced.

Reply:
It does not. Updated the policy definition.